crates/sargon-uniffi/src/system/drivers/secure_storage_driver/secure_storage_driver.rs

```diff
@@ -19,6 +19,11 @@ pub trait SecureStorageDriver: Send + Sync + std::fmt::Debug {
     ) -> Result<()>;
 
     async fn delete_data_for_key(&self, key: SecureStorageKey) -> Result<()>;
+
+    async fn contains_data_for_key(
```

Comment:
maybe a *batch* query would make sense? `contains_data_for_keys` ? Check with @micbakos-rdx but on iOS we can load the entire contents of Keychain in one go without prompting for biometrics and check the list of key-values. So we can actually impl `check_existence_of_mnemonics_for_device_factor_sources(ids: IndexSet<FactorSourceIDFromHash>) -> IndexSet<FactorSourceIDFromHash>` where if return value == ids.len() then **all** was found.

Reply:
for the current implementation we will only make the request for a given Factor Source, but I can extend it to take a set of keys instead.

Reply:
@micbakos-rdx mentioned that we can check for key existence without any biometrics.